src/KodKod.Analyzer/ApiActionsAreAttributeRoutedFixProvider.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace KodKod.Analyzer
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(ApiActionsAreAttributeRoutedFixProvider)), Shared]
+    public class ApiActionsAreAttributeRoutedFixProvider : CodeFixProvider
+    {
+        private const string HttpGetAttribute = "Microsoft.AspNetCore.Mvc.HttpGetAttribute";
+        private const string HttpPostAttribute = "Microsoft.AspNetCore.Mvc.HttpPostAttribute";
+        private const string HttpPutAttribute = "Microsoft.AspNetCore.Mvc.HttpPutAttribute";
+        private const string HttpDeleteAttribute = "Microsoft.AspNetCore.Mvc.HttpDeleteAttribute";
+
+        private static readonly Dictionary<string, string> Attributes = new Dictionary<string, string>
+        {
+            { "HttpGet", HttpGetAttribute },
+            { "HttpPost", HttpPostAttribute },
+            { "HttpPut", HttpPutAttribute },
+            { "HttpDelete", HttpDeleteAttribute },
+        };
+
+        private static readonly Dictionary<string, string> KeyWordMappings = new Dictionary<string, string>(StringComparer.Ordinal)
+        {
+            { "Get", HttpGetAttribute },
+            { "Find", HttpGetAttribute },
+            { "Post", HttpPostAttribute},
+            { "Put", HttpPutAttribute },
+            { "Delete", HttpDeleteAttribute },
+        };
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds =>
+            ImmutableArray.Create(DiagnosticDescriptors.KK1000_ApiActionsMustBeAttributeRouted.Id);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics[0];
```

Comment:
What if there are multiple? Guaranteed to be a at least one?

It would be good to check or assert this since the code is disconnected